### PR TITLE
A few small changes to move towards empty match files for CL

### PR DESCRIPTION
**Pull request by aarongreig**
* Return UNSUPPORTED_ENUMERATION in device info query when FP16
  extension isn't supported
* Add a mapping from CL_INVALID_ARG_SIZE to
  UR_RESULT_ERROR_INVALID_KERNEL_ARGUMENT_SIZE
* Remove a bunch of {{OPT}} entries from the enqueue match file, these
  should all be passing since bounds checking was added to the
  validation layer
* Write in a special case for PROGRAM_INFO_BINARIES test - this query
  works pretty differently to the others

LLVM testing https://github.com/intel/llvm/pull/12558

### CODE REVIEW

**Review by kbenzie — APPROVED** [review]

**Review by callumfare — APPROVED** [review]

File: test/conformance/program/urProgramGetInfo.cpp [hunk 1]
```diff
@@ -25,13 +25,31 @@ UUR_TEST_SUITE_P(
...
+                                        0, nullptr, &binary_sizes_len));
+        // Due to how the fixtures + env are set up we should only have one
+        // device associated with program, so one binary.
+        ASSERT_EQ(binary_sizes_len / sizeof(size_t), 1);
+        size_t binary_sizes[1] = {binary_sizes_len};
+        ASSERT_SUCCESS(urProgramGetInfo(program, UR_PROGRAM_INFO_BINARY_SIZES,
+                                        binary_sizes_len, binary_sizes,
+                                        nullptr));
+        std::vector<char> binary(binary_sizes[0]);
+        char *binaries[1] = {binary.data()};
```
Comment (omarahmed1111) [inline test/conformance/program/urProgramGetInfo.cpp:41]: I think we could check that the pointer returned is the same as the binary pointer that we initialized the program with

**Review by omarahmed1111 — APPROVED** [review]

File: test/conformance/program/urProgramGetInfo.cpp (hunk 1 above)
Comment (aarongreig) [reply test/conformance/program/urProgramGetInfo.cpp:41]: This is for querying out the device binary, I'll add something like that for `GET_IL` in https://github.com/oneapi-src/unified-runtime/pull/1196. I started adding a bit to this test where we feed the binary that was returned back into `urProgramCreateWithBinary` and then query the binary out of the new program and verify it against the binary we passed in, but then it occurred to me that `GET_PROGRAM_BINARIES` is very implementation dependant. The opencl equivalent of the query says the binary you get out is completely implementation defined, so as far as we know a given CL implementation might create a program with a binary that contains slightly different bits if it was created with `CreateWithBinary` and then built with `ProgramBuild`. There's no guarantee of consistency.

Comment (omarahmed1111) [reply test/conformance/program/urProgramGetInfo.cpp:41]: That make sense. Thanks for clarification.

**Comment by aarongreig** [comment]
merged into https://github.com/oneapi-src/unified-runtime/pull/1212

**Pull request closed without merging**